modules/base/src/processors/layercombiner.cpp

```diff
@@ -0,0 +1,187 @@
+        }
+
+        NumericType type = formats.front()->getNumericType();
+        for (auto f : formats) {
+            if (type == NumericType::Float || f->getNumericType() == NumericType::Float) {
+                type = NumericType::Float;
+            } else if (type == NumericType::SignedInteger ||
+                       f->getNumericType() == NumericType::SignedInteger) {
+                type = NumericType::SignedInteger;
+            } else if (f->getNumericType() == NumericType::UnsignedInteger) {
+                type = NumericType::UnsignedInteger;
+            }
+        }
+        size_t precision = (*std::ranges::max_element(formats, [](auto format1, auto format2) {
+                               return format1->getPrecision() < format2->getPrecision();
+                           }))->getPrecision();
+        return std::make_pair(type, precision);
+    }();
+
+    auto layer = std::make_shared<Layer>(*activePorts.front().first->getData(), noData,
+                                         DataFormatBase::get(type, activePorts.size(), precision));
+
+    auto layerRam = layer->getEditableRepresentation<LayerRAM>();
+
+#include <warn/push>
+#include <warn/ignore/conversion>
+#include <warn/ignore/conversion-loss>
+
+    layerRam->dispatch<void>([&](auto layerram) {
+        using PrecisionType = util::PrecisionValueType<decltype(layerram)>;
+        using ValueType = util::value_type_t<PrecisionType>;
+
+        const size2_t dims{layerram->getDimensions()};
+        auto destData = layerram->getDataTyped();
+
+        for (size_t inputChannel = 0; inputChannel < activePorts.size(); ++inputChannel) {
+            auto&& [port, srcChannel] = activePorts[inputChannel];
+            port->getData()->getRepresentation<LayerRAM>()->dispatch<void>(
+                [&](auto srclayer, int srcChannel) {
+                    const auto srcData = srclayer->getDataTyped();
+
+                    for (size_t i = 0; i < glm::compMul(dims); ++i) {
+                        util::glmcomp(destData[i], inputChannel) =
+                            static_cast<ValueType>(util::glmcomp(srcData[i], srcChannel));
+                    }
+                },
+                srcChannel);
+        }
+    });
+
+#include <warn/pop>
+
+    auto swizzleMask = [](size_t numComponents) {
+        switch (numComponents) {
+            case 1:
+                return swizzlemasks::luminance;
+            case 2:
+                return SwizzleMask{ImageChannel::Red, ImageChannel::Green, ImageChannel::Zero,
+                                   ImageChannel::One};
+            case 3:
+                return swizzlemasks::rgb;
+            case 4:
+            default:
+                return swizzlemasks::rgba;
+        }
+    };
+    layer->setSwizzleMask(swizzleMask(activePorts.size()));
+    layer->dataMap.dataRange = dataRange_.getDataRange();
+    layer->dataMap.valueRange = dataRange_.getValueRange();
```

Comment:
This will not work correctly in most cases. Probably will need to compute the min/max values

Reply:
At least this provides the opportunity to provide a custom range. If not used, the range of the first layer is used instead.
If you find an easy way to do this, you are welcome. The `VolumeCombiner` parses an equation, but that limits the functionality quite a bit.